packages/ra-ui-materialui/src/input/ReferenceArrayInput.spec.tsx

```diff
@@ -244,4 +249,52 @@ describe('<ReferenceArrayInput />', () => {
             });
         });
     });
+
+    it('should show selected values when ids type are inconsistant', async () => {
+        render(<HandlingTypesDiscrepencies />);
+        await waitFor(() => {
+            expect(
+                screen.queryByText('#1', {
+                    selector: 'div.MuiChip-root .MuiChip-label',
+                })
+            ).not.toBeNull();
+        });
+        expect(
+            screen.queryByText('#2', {
+                selector: 'div.MuiChip-root .MuiChip-label',
+            })
+        ).not.toBeNull();
+        expect(
+            screen.queryByText('artist_3', { selector: 'div.MuiChip-root' })
+        ).toBeNull();
+    });
+
+    it('should unselect a value whose id type is inconsistant', async () => {
+        render(<HandlingTypesDiscrepencies />);
+
+        const unselect = (queriedText: string) => () => {
+            const chip = screen.queryByText(queriedText, {
+                selector: '.MuiChip-label',
+            })?.nextSibling;
+            fireEvent.click(chip);
+        };
+
+        await waitFor(unselect('#1'));
+        await waitFor(unselect('#2'));
```

Comment:
I'm not fond of the way you wrote this. To me you should 
1. wait for the element to be ready with `await screen.findByXXX()`
2. then call `unselect` on it, without the need for `waitFor`

`waitFor` should be reserved for assertions and cases that cannot be covered by `await screen.findByXXX()`. Using it to wait for an action not to fail (which will lead to the action being attempted multiple times) is a bad practice IMHO.